### ISSUE TITLE
[0.3b2] Occasional 0 altitude issues, mostly just 10 hours of logs.

### ISSUE THREAD

**Issue by egid** [cyoung/stratux#54]
[stratux-uat.log - 107MB](https://drive.google.com/file/d/0B3DKMWgF5KKtdEpzbWRndDlxRzg/view?usp=sharing)

Took a trip from Seattle to Sacramento this weekend, about 5 hours each way. Had a couple instances of the 0 altitude issue; I also had an issue where ownship would get stuck at 4000-odd feet, which screws up nearby traffic altitudes (at 8500, aircraft on the ground at 500msl show as -35, ish).

Otherwise, worked fantastically throughout. Only really lost ADS-B briefly north of Portland; the rest of the trip we had 1-5 towers in view. GPS was pretty much rock solid, even with an unconfigured 1hz rate!

**Comment by cyoung** [comment]
I see 11 startups logged here - did you have to restart it for having locked up at all?

![ppm](https://cloud.githubusercontent.com/assets/57193/10010454/ac2cff12-60b9-11e5-9dc1-8225d100dd56.png)
![qos](https://cloud.githubusercontent.com/assets/57193/10010455/ad9050d4-60b9-11e5-9642-f6d4b490d9a6.png)

When you got the 0 relative altitude issue, did you do anything specific to resolve it? Force close, etc. Anything special seem to work?

Looks like there were only a handful (5-6) intervals of 5 minutes where you went without the latest METARs (QoS graph) within 100nm.

**Comment by egid** [comment]
I didn't have any lockups. I did restart it once to clear the 0 alt issue, and there were a few tests from the hotel in Reno just to see if I could get a tower. 

0 alt was fixed a few ways: cycling wifi on and off (had to wait for ADS-B to drop off); cycling power on and off (both iPads had the issue and I was lazy); force-quitting ForeFlight and relaunching (least-good option, loses route progress). Best solution seems to be cycling iPad wifi, but I didn't try it more than maybe twice so I'm not sure if it's a magic bullet.
